src/main/java/com/redhat/devtools/intellij/commonUiTestLibrary/fixtures/dialogs/FlatWelcomeFrame.java

```diff
@@ -63,17 +70,20 @@ public void createNewProject() {
      */
     public void clearWorkspace() {
         // delete all the projects' links from the 'Welcome to IntelliJ IDEA' dialog
-        int numberOfLinks = getNumberOfProjectLinks();
-        for (int i = 0; i < numberOfLinks; i++) {
-            ComponentFixture recentProjectsList = find(ComponentFixture.class, byXpath("//div[@accessiblename='Recent Projects' and @class='MyList']"), Duration.ofSeconds(10));
+        do {
+            List<JListFixture> jListFixtures = jLists(byXpath("//div[@accessiblename='Recent Projects']"));
+            if (jListFixtures.isEmpty() || jListFixtures.get(0).findAllText().size() == 1) {
+                break;
+            }
+            JListFixture recentProjectsList = jListFixtures.get(0);
             recentProjectsList.runJs("const horizontal_offset = component.getWidth()-22;\n" +
                     "robot.click(component, new Point(horizontal_offset, 22), MouseButton.LEFT_BUTTON, 1);");
             // Code for IntelliJ Idea 2020.3 or newer
             if (intelliJVersion.toInt() >= 20203) {
-                JPopupMenuFixture contextMenu = find(JPopupMenuFixture.class, JPopupMenuFixture.Companion.byType(), Duration.ofSeconds(10));
+                JPopupMenuFixture contextMenu = jPopupMenus(JPopupMenuFixture.Companion.byType()).get(0);
                 contextMenu.select("Remove from Recent Projects");
             }
-        }
+        } while (true);
```

Comment:
@zcervink I would prefer to rewrite this condition or even the loop to something like do-while (condition), instead of infinite loop with a break to have the code more readable.

Reply:
Fixed.